plugins/woocommerce/tests/e2e-pw/tests/merchant/create-coupon.spec.js

```diff
@@ -25,12 +25,17 @@ test.describe( 'Add New Coupon Page', () => {
 	} );
 
 	test( 'can create new coupon', async ( { page } ) => {
-		await page.goto( 'wp-admin/post-new.php?post_type=shop_coupon' );
+		await page.goto( 'wp-admin/post-new.php?post_type=shop_coupon', {
+			waitUntil: 'networkidle',
+		} );
+
 		await page.fill( '#title', couponCode );
+
 		await page.fill( '#woocommerce-coupon-description', 'test coupon' );
 
 		await page.fill( '#coupon_amount', '100' );
 
+		await expect( page.locator( '#publish:not(.disabled)' ) ).toBeVisible();
```

Comment:
One cause of flakiness in slower sites is that sometimes, the Publish button was still disabled by the time Playwright gets to this part. Playwright doesn't recognize the button as disabled for some reason, and therefore will successfully register a click. This line will assert the button to be enabled first, before performing the click.

Reply:
This issue has been reoccurring in the CD. Thanks for finally squashing this bug!